examples/tutorial/x/blog/handlers.go

```diff
@@ -0,0 +1,88 @@
+package blog
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+const costPer1000Chars = 1
+
+// RegisterRoutes will instantiate and register
+// all handlers in this package
+func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
+	r.Handle(PathCreatePostMsg, CreatePostMsgHandler{auth, NewPostBucket(), NewBlogBucket()})
+}
+
+type CreatePostMsgHandler struct {
+	auth  x.Authenticator
+	posts PostBucket
+	blogs BlogBucket
+}
+
+var _ weave.Handler = CreatePostMsgHandler{}
+
+func (h CreatePostMsgHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (weave.CheckResult, error) {
+	var res weave.CheckResult
+	msg, _, err := h.validate(ctx, db, tx)
+	if err != nil {
+		return res, err
+	}
+
+	res.GasAllocated = int64(len(msg.Text) / costPer1000Chars)
+	return res, nil
+}
+
+func (h CreatePostMsgHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (weave.DeliverResult, error) {
+	var res weave.DeliverResult
+	msg, blog, err := h.validate(ctx, db, tx)
+	if err != nil {
+		return res, err
+	}
+
+	post := &Post{
+		Title:  msg.Title,
+		Author: msg.Author,
+		Text:   msg.Text,
+	}
+
+	blog.NumArticles++
+	obj := orm.NewSimpleObj([]byte(post.Title), post) // Need to combine with count
```

Comment:
I would replace `[]byte(post.Title)` with `postKey(blog, post)` or something like that.

As mentioned on slack, would be cool if postKey is a composite key of (blog.Slug, blog.NumArticles) or such. You can add secondary indexes for author or even title if you wish (though searches are matches... not free-text searches, not so useful for title).